connectd/tor.c

```diff
@@ -41,6 +41,32 @@ struct connecting_socks {
 	struct connecting *connect;
 };
 
+static const char* socks5strerror(const tal_t *ctx, u8 code)
+{
+	// Error codes defined in https://tools.ietf.org/html/rfc1928#section-6
+	switch (code) {
+	case 0:
+		return tal_strdup(ctx, "success");
+	case 1:
+		return tal_strdup(ctx, "general SOCKS server failure");
+	case 2:
+		return tal_strdup(ctx, "connection not allowed by ruleset");
+	case 3:
+		return tal_strdup(ctx, "network unreachable");
+	case 4:
+		return tal_strdup(ctx, "host unreachable");
+	case 5:
+		return tal_strdup(ctx, "connection refused");
+	case 6:
+		return tal_strdup(ctx, "TTL expired");
+	case 7:
+		return tal_strdup(ctx, "command not supported");
+	case 8:
+		return tal_strdup(ctx, "address type not supported");
+	}
+	return tal_fmt(ctx, "unknown error: %d", code);
```

Comment:
just some minor nit, it's OK to int from IS0/IEC 9899:1999 perspective and its just a questions, but why not add an empty` default:`
for readability? :smile:

Reply:
>`%"PRIu8` might be  ̶c̶o̶r̶r̶e̶c̶t̶ pefered here instead of `%d` ? 

Good point.

Reply:
> just some minor nit, it's OK to int from IS0/IEC 9899:1999 perspective and its just a questions, but why not add an empty` default:`
> for readability?

It'd definitely oppose that, since it suppresses the compiler warning about not exhaustively matching all cases. If we were for example to add a case in the `enum` (not likely in this case, but still), the compiler would warn us that there is an unhandled case in the `switch` statement. Without that warning it is likely that we'd wait for a user to complain about `unknown error: xyz` and only then add `case xyz` to the switch statement.

Reply:
But the latest code does not use an `enum` anymore, rather a direct `u8`,so that specific objection does not apply.